pkg/operators/ebpf/ebpf.go

```diff
@@ -362,6 +375,40 @@ func (i *ebpfInstance) register(gadgetCtx operators.GadgetContext) error {
 		m.accessor = accessor
 		m.ds = ds
 	}
+	for name, m := range i.mapIters {
+		fields := make([]*Field, 0)
+		ds, err := gadgetCtx.RegisterDataSource(datasource.TypeArray, name)
+		if err != nil {
+			return fmt.Errorf("adding mapiter datasource: %w", err)
+		}
+		ds.AddAnnotation("fetch-count", fmt.Sprintf("%d", m.count))
+		ds.AddAnnotation("fetch-interval", m.interval.String())
```

Comment:
This can't go here because the map params are evaluated in `evaluateMapParams()`, which is called after `register()`.